tests/transform/test_model_details.py

```
@@ -130,6 +139,7 @@ class TestTransformModelDetails:
                 "allele_info": "model_details_allele_info_good_test_input.csv",
                 "model_info": "model_details_model_info_missing_column_input.csv",
                 "pathology": "model_details_pathology_good_test_input.csv",
+                "model_results_info": "model_details_model_results_info_good_test_input.csv",
```

Could you add a test for when the model details file _doesn't_ contain the model referenced by the rest of the files? This actually seems likely to happen: if we add data for a new model and accidentally run this before updating the model details file.

---

Oh, I see you might already be testing this with the "hTau" model. But I think the code should raise an error if we're trying to get data for a model that isn't in the model details file, instead of silently just putting nulls. @JessterB what do you think?

---

I think your question is:
If we are building `model_details` for a model that isn't flagged to include any of the three links, should we throw an error?

We will sometimes have models with only pathology and biomarker data (which aren't represented by links here), and we may also start introducing 'emerging models' that do not yet have results of any type at all.

Since there are valid cases where all three of these fields will be null, I don't think throwing an error would be helpful.

---

No, my question is whether we should throw an error if the model doesn't have a line _at all_ in the model_details input file. If a model has only biomarkers and pathology, it should still have a line in that file even if the three links fields are empty. I'm thinking of the case where we are missing a model entirely from the file.

---

There are two different model_details input files:

* model_info (https://www.synapse.org/Synapse:syn61378590)
Specifies general model metadata values like name, jax_id, and so on. If a model does not have a row in model_info, we should not be trying to generate a model_details object for it.

* model_results_info (https://www.synapse.org/Synapse:syn68377668)
Specifies which result types we have for a model. If a model does not have a row in model_results_info, we could either:
1. infer that there are no results for that model (yet) available 
2. throw an error

I'm not totally against 2, it would just mean that we'd need to add an empty row to model_results_info when we add an emerging model.

---

Ohhh I see. Thank you for clarifying that, I think I misunderstood part of how it worked. Then, we probably don't need to throw an error for this and can just assume it has no results yet, which is what Beatriz has it currently doing.